Library/DiscUtils.Ext/ExtentsFileBuffer.cs

```diff
@@ -124,69 +134,91 @@ public override IEnumerable<StreamExtent> GetExtentsInRange(long start, long cou
                 new StreamExtent(start, count));
         }
 
+        /// <summary>
+        /// Returns the first extent containing a logical block greater than or equal to the given one.
+        /// Returns null if no such extent exists.
+        /// </summary>
         private Extent FindExtent(ExtentBlock node, uint logicalBlock)
         {
-            if (node.Index != null)
+            if (node.Extents != null)
+            {
+                return FindExtent(node.Extents, logicalBlock);
+            }
+            else if (node.Index != null)
+            {
+                return FindExtent(node.Index, logicalBlock);
+            }
+            else
             {
-                ExtentIndex idxEntry = null;
+                return null;
+            }
+        }
 
-                if (node.Index.Length == 0)
-                {
-                    return null;
-                }
-                if (node.Index[0].FirstLogicalBlock >= logicalBlock)
-                {
-                    idxEntry = node.Index[0];
-                }
-                else
+        /// <summary>
+        /// Returns the first extent containing a logical block greater than or equal to the given one.
+        /// Returns null if no such extent exists.
+        /// </summary>
+        private Extent FindExtent(Extent[] extents, uint logicalBlock)
+        {
+            for (int i = 0; i < extents.Length; ++i)
+            {
+                Extent extent = extents[i];
+                if (extent.FirstLogicalBlock >= logicalBlock ||
+                    extent.FirstLogicalBlock + extent.NumBlocks > logicalBlock)
```

Comment:
Just wanted to confirm that change change is mainly refactoring, not real bug fix.

Reply:
As explained offline, this is part of the actual fix.